spacy_transformers/layers/hf_shim.py

```diff
@@ -24,8 +24,14 @@ def __init__(
         optimizer: Any = None,
         mixed_precision: bool = False,
         grad_scaler_config: dict = {},
+        config_cls = AutoConfig,
+        model_cls = AutoModel,
+        tokenizer_cls = AutoTokenizer,
     ):
         self._hfmodel = model
+        self.config_cls = config_cls
+        self.model_cls = model_cls
+        self.tokenizer_cls = tokenizer_cls
```

Comment:
They're not `Optional`, so checking for `None` doesn't sound very helpful to me, vs. checking something more relevant?

You can check that they have the right methods at the point where they're called, but if they don't, my thinking was that the error message should be pretty obvious and an additional check wouldn't really help much.

Reply:
I felt like it's the type of error that will occur most often - users thinking they're passing on some class that was determined earlier on in the code but the variable actually holding `None` accidentally. If we don't do the check, of course the exception will still be raised and it'll be up to the user to understand what happened, rather than us providing a more explicit error message straight away. We can merge this as is and revisit in case this comes up on the forum.